test/integration/ads_integration_test.cc

```diff
@@ -219,6 +219,67 @@ TEST_P(AdsIntegrationTest, ClusterSharingSecretWarming) {
   test_server_->waitForGaugeEq("cluster_manager.warming_clusters", 0);
 }
 
+// Make sure two two clusters with different secrets send only a single SDS request.
```

Comment:
nit, double `two` here

Reply:
> I'd suggest runtime guarding.

Added a runtime flag.
By default set to true, LMK if you think it should be the set to false.